dotCMS/src/main/java/com/dotcms/browser/BrowserAPI.java

```diff
@@ -0,0 +1,721 @@
+                                                final boolean onlyFiles, final String sortBy, final boolean sortByDesc, final boolean excludeLinks, final long languageId, final boolean dotAssets)
+			throws DotSecurityException, DotDataException {
+
+	    final boolean showPages =! onlyFiles;
+
+	    final BrowserQuery browserQuery = BrowserQuery.builder()
+	                    .showDotAssets(dotAssets)
+	                    .showLinks(!excludeLinks)
+	                    .showExtensions(extensions)
+	                    .withFilter(filter)
+	                    .withHostOrFolderId(folderId)
+	                    .withLanguageId(languageId)
+	                    .offset(offset)
+	                    .showFiles(true)
+	                    .showPages(showPages)
+	                    .showFolders(!noFolders)
+	                    .showArchived(showArchived)
+	                    .showWorking(showWorking)
+                        .showMimeTypes(mimeTypes)
+                        .maxResults(maxResults)
+	                    .sortBy(sortBy)
+	                    .sortByDesc(sortByDesc)
+	                    .withUser(user).build();
+	    
+	    return getFolderContent(browserQuery);
+	}
+	
+    @CloseDBIfOpened
+    public Map<String, Object> getFolderContent(final BrowserQuery browserQuery) throws DotSecurityException, DotDataException {
+
+        final boolean respectFrontEndRoles = false;
+        List<Map<String, Object>> returnList = new ArrayList<>();
+        final Role[] roles = APILocator.getRoleAPI().loadRolesForUser(browserQuery.user.getUserId()).toArray(new Role[0]);
+
+        // gets folder parent
+        final Folder parent = folderAPI.find(browserQuery.hostFolderId, browserQuery.user, respectFrontEndRoles);
+
+        Host host = null;
+        if (parent == null) {// If we didn't find a parent folder lets verify if
+                             // this is a host
+            host = APILocator.getHostAPI().find(browserQuery.hostFolderId, browserQuery.user, respectFrontEndRoles);
+
+            if (host == null) {
+
+                Logger.error(this, "Folder ID doesn't belong to a Folder nor a Host, id: " + browserQuery.hostFolderId
+                                + ", maybe the Folder was modified in the background.");
+                throw new NotFoundInDbException("Folder ID doesn't belong to a Folder nor a Host, id: " + browserQuery.hostFolderId);
+            }
+        }
+
+        if (browserQuery.showFolders) {
+
+            this.includeFolders(browserQuery, returnList, roles, parent);
+        }
+
+        if (browserQuery.showLinks) {
+
+            this.includeLinks(browserQuery, returnList, roles, parent, host);
+        }
+
+        final String luceneQuery = this.createQuery(browserQuery, parent, host);
+        final String esSortBy    = ("name".equals(browserQuery.sortBy) ? "title" : browserQuery.sortBy)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "title" appears 7 times in this file; the first occurrence is on line 347](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337824)